tools/api-markdown-documenter/src/documentation-domain/DocumentNode.ts

```diff
@@ -38,6 +38,31 @@ export interface DocumentNodeProperties {
 	readonly frontMatter?: string;
 }
 
+/**
+ * Metadata of a {@link DocumentNode} in terms of its API.
+ *
+ * @remarks
+ * `DocumentItemMetadata` aids in tracing a documentation node to its API, useful for cross-referencing and integrations.
+ *
+ * @public
+ */
+export interface DocumentItemMetadata {
+	/**
+	 * Name of the original API, e.g., class or function, from which this documentation node is derived.
+	 */
+	readonly apiItemName: string;
+
+	/**
+	 * Category or type of the API like 'class' or 'function'.
+	 */
+	readonly apiItemKind: string;
```

Comment:
We should be able to use the `ApiItemKind` enum from `@microsoft/api-extractor-model` here. It's actually a fixed set of things. 

If it isn't already there, we should also add `ApiItemKind` to our "convenience re-exports" section of the package's root `index.ts` file.